src/main/java/com/conveyal/r5/util/TIntObjectHashMultimap.java

```diff
@@ -17,7 +17,12 @@
 
     @Override
     public boolean put(int key, V value) {
-        if (!wrapped.containsKey(key)) wrapped.put(key, new ArrayList<>());
+        if (!wrapped.containsKey(key)) {
+            wrapped.put(key, new ArrayList<>(2));
+        } else {
+            ArrayList<V> wrappedList = (ArrayList<V>) wrapped.get(key);
+            wrappedList.ensureCapacity(wrappedList.size() * 2);
```

Comment:
good idea with `ensureCapacity` but I wonder if it would be better to add some constant value here instead of multiplying it by 2. I am not sure how large this map would be but in case of a large one we would have more allocations but less memory overall.

Reply:
Hi, some comments on this change as submitted to the R5 library: https://github.com/conveyal/r5/pull/864#issuecomment-1469494133